chainer/functions/array/broadcast.py

```diff
@@ -108,11 +111,9 @@ def check_type_forward(self, in_types):
             actual = 'in_type[0].shape: %s' % str(shape)
             raise type_check.InvalidType(expect, actual)
 
-    def forward(self, xs):
-        self.retain_inputs(())
-        xp = cuda.get_array_module(*xs)
-        x = xs[0]
-        self._xp = xp
+    def forward(self, inputs):
+        x, = inputs
+        xp = cuda.get_array_module(x)
         self._in_shape = x.shape
```

Comment:
You can fetch input shape/dtype from `self.inputs[0]` (which is a `VariableNode` object) in backward even if the input array is not retained.